app/browser-detection.js

```diff
@@ -0,0 +1,226 @@
+        [ 'Windows 7', /(Windows NT 6.1)/ ],
+        [ 'Windows 8', /(Windows NT 6.2)/ ],
+        [ 'Windows 8.1', /(Windows NT 6.3)/ ],
+        [ 'Windows 10', /(Windows NT 10.0)/ ],
+        [ 'Windows ME', /Windows ME/ ],
+        [ 'Open BSD', /OpenBSD/ ],
+        [ 'Sun OS', /SunOS/ ],
+        [ 'Linux', /(Linux)|(X11)/ ],
+        [ 'Mac OS', /(Mac_PowerPC)|(Macintosh)/ ],
+        [ 'QNX', /QNX/ ],
+        [ 'BeOS', /BeOS/ ],
+        [ 'OS/2', /OS\/2/ ],
+        [ 'Search Bot', /(nuhk)|(Googlebot)|(Yammybot)|(Openbot)|(Slurp)|(MSNBot)|(Ask Jeeves\/Teoma)|(ia_archiver)/ ]
+      ]);
+    }
+
+    function buildRules(ruleTuples) {
+      return ruleTuples.map(function(tuple) {
+        return {
+          name: tuple[0],
+          rule: tuple[1]
+        };
+      });
+    }
+
+    function renderNotSupported(){
+        var cqNotSupported = document.createElement('div');
+        cqNotSupported.className = 'cq-not-supported';
+
+        var cqLogo = document.createElement('div');
+        cqLogo.className = 'cq-logo';
+        cqNotSupported.appendChild(cqLogo);
+
+        var cqIconNotSupported = document.createElement('div');
+        cqIconNotSupported.className = 'cq-icon-not-supported';
+        cqNotSupported.appendChild(cqIconNotSupported);
+
+        var cqH1 = document.createElement('h1');
+        cqH1.innerHTML = 'YOUR BROWSER IS NOT SUPPORTED';
+        cqNotSupported.appendChild(cqH1);
+
+        var cqP = document.createElement('p');
+        cqP.innerHTML = 'Please update your browser for the best trading experience.';
+        cqNotSupported.appendChild(cqP);
+
+        var cqA = document.createElement('a');
+        cqA.href = 'http://outdatedbrowser.com/';
+        cqA.innerHTML = 'Update Browser';
+        cqNotSupported.appendChild(cqA);
+        
+        if (cqRoot.nextSibling) {
+            cqRoot.parentNode.insertBefore(cqNotSupported, cqRoot.nextSibling);
+        } else {
+            cqRoot.parentNode.appendChild(cqNotSupported);
+        }
+        cqRoot.parentNode.removeChild(cqRoot);
+    }
+
+    window.onerror = function(message, source, lineno, colno, error){
+        if (
+                (
+                    message.indexOf('[mobx] MobX 5+ requires Proxy objects') !== -1 ||
+                    message.indexOf('not supported browser') !== -1
+                ) && cqRoot
+            ) {
+            renderNotSupported();
+        }
+    }
+
+    var browser = detect();
+    if (browser) {
+        var version = parseInt(browser.version.split('.').join(''));
+        var name = browser.name;
+        var os = browser.os;
+
+        if (
+            SupportedBrowsers[os] && 
+            SupportedBrowsers[os][name] !== undefined) {
+            if(version < SupportedBrowsers[os][name]){
+                throw new Error('not supported browser');
+            }
+        } else {
+            alert('Please update your browser to the latest version.');
```

Comment:
because I don't have access to `notification` component so the only way is to `alert` the message

Reply:
@shayanbinary why are you using `alert`? We have a design from the UX team to display some page is the browser is not supported. Use that instead